blueprints/data-solutions/shielded-folder/variables.tf

```diff
@@ -0,0 +1,232 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# tfdoc:file:description Variables definition.
+
+variable "access_policy" {
+  description = "Access Policy name, set to null if creating one."
+  type        = string
+  default     = null
+}
+
+variable "access_policy_create" {
+  description = "Access Policy configuration, fill in to create. Parent is in 'organizations/123456' format."
+  type = object({
+    parent = string
+    title  = string
+    scopes = optional(list(string))
+  })
+  default = null
+}
+
+variable "data_dir" {
+  description = "Relative path for the folder storing configuration data."
+  type        = string
+  default     = "data"
+}
+
+variable "enable_features" {
+  description = "Flag to enable features on the solution."
+  type = object({
+    encryption = bool
+    log_sink   = bool
+    vpc_sc     = bool
+  })
+  default = {
+    encryption = false
+    log_sink   = true
+    vpc_sc     = true
+  }
+}
+variable "folder_create" {
+  description = "Provide values if folder creation is needed, uses existing folder if null. Parent is in 'folders/nnn' or 'organizations/nnn' format."
+  type = object({
+    display_name = string
+    parent       = string
+  })
+  default = null
+}
+
+variable "folder_id" {
+  description = "Folder ID in case you use folder_create=null."
+  type        = string
+  default     = null
+}
+
+variable "groups" {
+  description = "User groups."
+  type        = map(string)
+  default = {
+    workload-engineers = "gcp-data-engineers"
```

Comment:
please use a strict type and optional with defaults here to allow overriding individual values

```hcl
type = object({
  workload-engineers = optional(string, "gcp-data-engineers")
  workload-security = optional(string, "gcp-data-security")
})
default = {}
nullable = false
```